### ISSUE TITLE
feat(training): FastTD3Trainer provider — TD3 + CDQ + C51 critic + AMP-bf16 + torch.compile

### ISSUE THREAD

**Issue by cagataycali** [strands-labs/robots#862]
Part of #858 epic. Depends on #C1 (VecEnv), #C2 (Replay), #C3 (modules).

## Goal
Land `strands_robots.training.fasttd3.FastTD3Trainer` so:

```python
trainer = create_trainer(\"fasttd3\")
spec = TrainSpec(
    env_name=\"Isaac-Velocity-Flat-G1-v0\",
    output_dir=\"/tmp/g1\",
    steps=100_000,
    extra={\"agent\": \"fasttd3\", \"num_envs\": 4096, \"batch_size\": 32768},
)
trainer.train(spec)
```

resolves through `registry/policies.json` with a `trainer` block (no
matching `Policy` is required — RL trainers are inference-free here).

## Algorithm features
- TD3 (Fujimoto+ 2018): twin Q, delayed policy update (`policy_frequency`),
  target policy smoothing (`policy_noise`, `noise_clip`).
- **Clipped Double Q-Learning** (`use_cdq=True`).
- **Distributional critic (C51)**: `num_atoms`, `v_min`, `v_max` per env preset.
- **AMP bf16 + torch.compile** (with `compile_mode=\"reduce-overhead\"` /
  `\"max-autotune\"` knobs).
- **Multi-step return** (`num_steps=1..N`, default 1, n=3 for some tasks).
- **Optional**: SimbaV2 actor/critic, SimNorm embeddings (gated via
  `--agent fasttd3_simbav2` / `--sim_type`).
- **Optional**: gradient norm clipping (`use_grad_norm_clipping`,
  `max_grad_norm`).
- **Optional**: bootstrap disabling (`disable_bootstrap` — needed for IsaacLab
  manipulation with brittle reward shaping).
- **`train_mode()` / `eval_mode()`** switching of obs normalizers.

## Hyperparam presets to port (from FastTD3 `hyperparams.py`)
- HumanoidBench: `h1hand-reach/balance/pole/maze/push/...`
- MuJoCo Playground: `G1JoystickFlatTerrain`, `T1JoystickFlatTerrain`,
  `LeapCubeReorient`, `LeapCubeRotateZAxis`, `Go1JoystickFlatTerrain`,
  `Go1Getup`, `CheetahRun`
- IsaacLab: `Isaac-Lift-Cube-Franka-v0`, `Isaac-Open-Drawer-Franka-v0`,
  `Isaac-Velocity-Flat-{H1,G1}-v0`, `Isaac-Velocity-Rough-{H1,G1}-v0`,
  `Isaac-Repose-Cube-{Allegro,Shadow}-Direct-v0`
- MTBench: `MetaWorldMT10/50`

Land them as a `strands_robots/training/_presets/fasttd3.json` so users get
benchmarked starting points.

## Acceptance
- [ ] `FastTD3Trainer` passes the same Trainer ABC contract
      (`validate → prepare → train → export`)
- [ ] `trainer.export()` produces a torch.jit.script-able policy (for #C13)
- [ ] CI smoke test: 1000-step training on a CartPole-class env converges
- [ ] Bench parity: G1 IsaacLab velocity-flat reaches FastTD3 paper's
      ~50k step milestone within 10% wall-time on the same hw

## Refs
- FastTD3: https://github.com/younggyoseo/FastTD3 (MIT, arXiv:2505.22642)
- Hyperparam catalog: `fast_td3/hyperparams.py`
- Main loop reference: `fast_td3/train.py`, `fast_td3/fast_td3.py`

**Comment by cagataycali** [comment]
Closing as out of scope for 1.0, and folding the tracking back into the parent epic #858.

The 0.5 -> 1.0 roadmap has six themes - tree-shaking, the dashboard, sealing the policy-provider list, a docs/examples rework, a starter kit and the Sprout embodiment. An off-policy RL training stack with parallel-sim backends (IsaacLab, MJWarp, MuJoCo Playground, HumanoidBench) is not among them, and it is a large enough body of work that keeping fifteen open children makes the 1.0 board unreadable.

Measured before closing, so this is not being written off as already-done: four of these children look shipped if you grade them by whether a class exists, and none of them meets its own acceptance criteria. `FastSacTrainer`, `PpoTrainer`, `VecSimEnv` and `elastic_launch_callable` are all present, but `num_atoms`/`c51`/`v_min` appear nowhere (so #859's distributional critic did not land), `adaptive_kl`/`desired_kl` appear nowhere (#865), `mjwarp` appears nowhere (so #863's vec-env does not span the three backends it names), and `elastic_launch_callable` has zero references from `training/rl/` (#870 asked for a launcher FOR the RL trainers). A symbol is a claim; the criteria are the contract.

#858 stays open as the single post-1.0 tracker for this stack. Re-file any individual piece against the 1.0+ roadmap when the RL lane is scheduled.

Closed as part of the 0.5 -> 1.0 backlog cleanup.
